modeshape-jcr-api/src/main/java/org/modeshape/jcr/api/value/DateTime.java

```diff
@@ -274,151 +187,21 @@
-
-    /**
-     * Subtract the specified number of years from this time instant.
-     * 
-     * @param years the number of years to subtract
-     * @return the instance in time the specified number of years before this instant
-     */
-    DateTime minusYears( int years );
+    DateTime minus( Duration duration );
 
     /**
-     * Add the specified about of time in the supplied units.
+     * Add the specified amount of time.
      * 
-     * @param timeAmount the amount of time to add
-     * @param unit the units of the amount of time; may not be null
+     * @param duration the amount of time; may not be null
      * @return the instance in time the specified number of time after this instant
      */
-    DateTime plus( long timeAmount,
-                   TimeUnit unit );
-
-    /**
-     * Add the specified number of days from this time instant.
-     * 
-     * @param days the number of days to add
-     * @return the instance in time the specified number of days after this instant
-     */
-    DateTime plusDays( int days );
-
-    /**
-     * Add the specified number of hours from this time instant.
-     * 
-     * @param hours the number of hours to add
-     * @return the instance in time the specified number of hours after this instant
-     */
-    DateTime plusHours( int hours );
-
-    /**
-     * Add the specified number of milliseconds from this time instant.
-     * 
-     * @param milliseconds the number of milliseconds to add
-     * @return the instance in time the specified number of milliseconds after this instant
-     */
-    DateTime plusMillis( int milliseconds );
-
-    /**
-     * Add the specified number of minutes from this time instant.
-     * 
-     * @param minutes the number of minutes to add
-     * @return the instance in time the specified number of minutes after this instant
-     */
-    DateTime plusMinutes( int minutes );
-
-    /**
-     * Add the specified number of months from this time instant.
-     * 
-     * @param months the number of months to add
-     * @return the instance in time the specified number of months after this instant
-     */
-    DateTime plusMonths( int months );
-
-    /**
-     * Add the specified number of seconds from this time instant.
-     * 
-     * @param seconds the number of seconds to add
-     * @return the instance in time the specified number of seconds after this instant
-     */
-    DateTime plusSeconds( int seconds );
-
-    /**
-     * Add the specified number of weeks from this time instant.
-     * 
-     * @param weeks the number of weeks to add
-     * @return the instance in time the specified number of weeks after this instant
-     */
-    DateTime plusWeeks( int weeks );
-
-    /**
-     * Add the specified number of years from this time instant.
-     * 
-     * @param years the number of years to add
-     * @return the instance in time the specified number of years after this instant
-     */
-    DateTime plusYears( int years );
```

Comment:
>Should we handle changing public APIs like this as separate issues?

Ideally yes. But in practice after changing the JDK and jboss-parent I started fixing issues incrementally, so it was really had to split each change. I _could_ try to do it, but it would be probably hard since there are a lot of intersecting issues.
For example: fixing javadoc and compiler warnings, code style refactorings and actual changing code may affect the same file.

> Besides, couldn't all of these be implemented as default methods by using toZonedDateTime (or toLocalDateTime)?

Yes, but these particular methods were already used throughout the existing code, so keeping those methods as they were was easier than changing a lot of other places.